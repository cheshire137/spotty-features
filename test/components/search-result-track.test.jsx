import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import TrackSearchResponse from '../fixtures/spotify/track-search'

import SearchResultTrack from '../../src/components/search-result-track.jsx'

const track = TrackSearchResponse.tracks.items[0]
const image = track.album.images[2].url

function props(chooseTrack, select) {
  return {
    artists: track.artists.map(a => a.name),
    album: track.album.name,
    image,
    name: track.name,
    selected: false,
    chooseTrack,
    select
  }
}

describe('SearchResultTrack', () => {
  let component = null
  let trackChosen = false
  let trackSelected = false

  beforeEach(() => {
    const chooseTrack = () => {
      trackChosen = true
    }
    const select = () => {
      trackSelected = true
    }
    component = <SearchResultTrack {...props(chooseTrack, select)} />
  })

  test('matches snapshot', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('lists artists', () => {
    const artists = shallow(component).find('.track-artists')
    expect(artists.text()).toBe('Grimes, Aristophanes')
  })

  test('displays song title', () => {
    const title = shallow(component).find('.track-name')
    expect(title.text()).toBe('SCREAM')
  })

  test('displays album name', () => {
    const album = shallow(component).find('.track-album')
    expect(album.text()).toBe('Art Angels')
  })

  test('displays album art', () => {
    const images = shallow(component).find('.track-image')
    expect(images.length).toBe(1)
    expect(images.at(0).props().src).toBe(image)
  })

  test('search result can be chosen', () => {
    const button = shallow(component).find('.search-result-button')
    expect(trackChosen).toBe(false)
    button.simulate('click')
    expect(trackChosen).toBe(true)
  })

  test('search result can be selected', () => {
    const button = shallow(component).find('li')
    expect(trackSelected).toBe(false)
    button.simulate('mouseover')
    expect(trackSelected).toBe(true)
  })
})
