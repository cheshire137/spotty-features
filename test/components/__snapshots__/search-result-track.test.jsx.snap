exports[`SearchResultTrack matches snapshot 1`] = `
<li
  className=""
  onMouseOver={[Function]}>
  <button
    className="search-result-button"
    onClick={[Function]}
    type="button">
    <span
      className="columns">
      <span
        className="column track-image-column">
        <img
          className="track-image"
          src="https://i.scdn.co/image/b6a166bdf785f16e212ef95984fcf2e788678597" />
      </span>
      <span
        className="column search-result-details">
        <span
          className="track-name">
          SCREAM
        </span>
        <span>
          by
        </span>
        <span
          className="track-artists">
          Grimes, Aristophanes
        </span>
        <span
          className="track-album">
          Art Angels
        </span>
      </span>
    </span>
  </button>
</li>
`;
